sklearn/decomposition/_dict_learning.py

```diff
@@ -861,218 +802,34 @@ def dict_learning_online(
     SparsePCA : Sparse Principal Components Analysis.
     MiniBatchSparsePCA : Mini-batch Sparse Principal Components Analysis.
     """
-    deps = (return_n_iter, return_inner_stats, iter_offset, inner_stats)
-    if max_iter is not None and not all(arg == "deprecated" for arg in deps):
-        raise ValueError(
-            "The following arguments are incompatible with 'max_iter': "
-            "return_n_iter, return_inner_stats, iter_offset, inner_stats"
-        )
-
-    iter_offset = _check_warn_deprecated(iter_offset, "iter_offset", default=0)
-    return_inner_stats = _check_warn_deprecated(
-        return_inner_stats,
-        "return_inner_stats",
-        default=False,
-        additional_message="From 1.4 inner_stats will never be returned.",
-    )
-    inner_stats = _check_warn_deprecated(inner_stats, "inner_stats", default=None)
-    return_n_iter = _check_warn_deprecated(
-        return_n_iter,
-        "return_n_iter",
-        default=False,
-        additional_message=(
-            "From 1.4 'n_iter' will never be returned. Refer to the 'n_iter_' and "
-            "'n_steps_' attributes of the MiniBatchDictionaryLearning object instead."
-        ),
-    )
-
-    if max_iter is not None:
-        transform_algorithm = "lasso_" + method
-
-        est = MiniBatchDictionaryLearning(
-            n_components=n_components,
-            alpha=alpha,
-            n_iter=n_iter,
```

Comment:
This is weird here. we never passed `max_iter` while I assume we would have expected to do so.

Reply:
Yes it was a mistake. We should have passed max_iter instead of n_iter

Reply:
I will had an entry in the what's new to acknowledge that.